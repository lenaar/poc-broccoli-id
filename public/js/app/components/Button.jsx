import React, { useState } from 'react'

import i18n from '../../../../i18n'

function Button({ caption = 'N/A', lang = 'sv', onClick = null }) {
  const [buttonClicked, setButtonClicked] = useState(false)
  const doClick = onClick || setButtonClicked
  return (
    <>
      <button type="button" className="btn btn-primary" onClick={() => doClick(true)}>
        {caption}
      </button>
      {buttonClicked ? <p>{i18n.message('template_button_works', lang)}</p> : null}
    </>
  )
}

export default Button
