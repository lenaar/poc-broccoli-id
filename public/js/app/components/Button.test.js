import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Button from './Button'

describe('Component <Button>', () => {
  test('creates a button', () => {
    const { asFragment } = render(<Button caption="This is a message" />)

    expect(asFragment()).toMatchSnapshot()
    // or
    expect(screen.getByText('This is a message')).toHaveAttribute('type', 'button')
    expect(() => screen.getByText('Knappen fungerar!')).toThrow()
  })

  test('changes output after first click', () => {
    const { asFragment } = render(<Button caption="This is a message" />)
    const stateBeforeFirstClick = asFragment()

    fireEvent.click(screen.getByText('This is a message'))
    expect(asFragment()).not.toEqual(stateBeforeFirstClick)

    expect(asFragment()).toMatchSnapshot()
    // or
    expect(screen.getByText('This is a message')).toHaveAttribute('type', 'button')
    expect(screen.getByText('Knappen fungerar!'))
  })

  test("doesn't change anything after another click", () => {
    const { asFragment } = render(<Button caption="This is a message" />)

    fireEvent.click(screen.getByText('This is a message'))
    const stateAfterFirstClick = asFragment()

    fireEvent.click(screen.getByText('This is a message'))
    expect(asFragment()).toEqual(stateAfterFirstClick)
  })
})
