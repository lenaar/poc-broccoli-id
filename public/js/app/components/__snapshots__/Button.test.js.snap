// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Button> changes output after first click 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
    type="button"
  >
    This is a message
  </button>
  <p>
    Knappen fungerar!
  </p>
</DocumentFragment>
`;

exports[`Component <Button> creates a button 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
    type="button"
  >
    This is a message
  </button>
</DocumentFragment>
`;
